client/src/TenderList.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+const TendersList = () => {
+	const { pageNumber } = useParams();
+	const currentPage = pageNumber ? parseInt(pageNumber, 10) : 1;
+	const [tenders, setTenders] = useState([]);
+	const [loading, setLoading] = useState(true);
+	const [pagination, setPagination] = useState({
+		itemsPerPage: 25,
+		currentPage: currentPage,
+		totalPages: 1,
+	});
+	const [error, setError] = useState(null);
+	const navigate = useNavigate();
+
+	const fetchTenders = async (page) => {
+		setLoading(true);
+	try {
+		const response = await fetch(`/api/tenders?page=${page}`);
+		if (!response.ok) {
+			throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
The user doesn't need to know the status code. Imagine how a user would react if they went to a page, and they saw `Status code: 500`. It doesn't tell them anything.